Samples/SwiftUITestbed/Sources/MainWorkflow.swift

```diff
@@ -39,53 +41,96 @@ struct MainWorkflow: Workflow {
-
-        case pushScreen
-        case presentScreen
-        case changeTitle(String)
-        case changeAllCaps(Bool)
-
-        func apply(toState state: inout WorkflowType.State) -> WorkflowType.Output? {
-            switch self {
-            case .pushScreen:
-                return .pushScreen
-            case .presentScreen:
-                return .presentScreen
-            case .changeTitle(let newValue):
-                state.title = newValue
-                state.isAllCaps = newValue.isAllCaps
-            case .changeAllCaps(let isAllCaps):
-                state.isAllCaps = isAllCaps
-                state.title = isAllCaps ? state.title.uppercased() : state.title.lowercased()
-            }
-            return nil
-        }
-    }
-
     typealias Rendering = MainScreen
+    typealias Action = MainScreen.Action
 
     func render(state: State, context: RenderContext<Self>) -> Rendering {
-        let sink = context.makeSink(of: Action.self)
+        let sink = state.trampoline.makeSink(of: Action.self, with: context)
 
         return MainScreen(
             title: state.title,
-            didChangeTitle: { sink.send(.changeTitle($0)) },
+            canClose: canClose,
             allCapsToggleIsOn: state.isAllCaps,
             allCapsToggleIsEnabled: !state.title.isEmpty,
-            didChangeAllCapsToggle: { sink.send(.changeAllCaps($0)) },
-            didTapPushScreen: { sink.send(.pushScreen) },
-            didTapPresentScreen: { sink.send(.presentScreen) },
-            didTapClose: didClose
+            sink: sink
         )
     }
 }
 
+extension MainScreen.Action: WorkflowAction {
+    typealias WorkflowType = MainWorkflow
+
+    func apply(toState state: inout WorkflowType.State) -> WorkflowType.Output? {
+        switch self {
+        case .pushScreen:
+            return .pushScreen
+        case .presentScreen:
+            return .presentScreen
+        case .changeTitle(let newValue):
+            state.title = newValue
+            state.isAllCaps = newValue.isAllCaps
+        case .changeAllCaps(let isAllCaps):
+            state.isAllCaps = isAllCaps
+            state.title = isAllCaps ? state.title.uppercased() : state.title.lowercased()
+        case .close:
+            return .close
+        }
+        return nil
+    }
+}
+
 private extension String {
     var isAllCaps: Bool {
         allSatisfy { character in
             character.isUppercase || !character.isCased
         }
     }
 }
+
+class SinkTrampoline: Equatable {
+    private var sinks: [ObjectIdentifier: Any] = [:]
+
+    func makeSink<Action, WorkflowType>(
+        of actionType: Action.Type,
+        with context: RenderContext<WorkflowType>
+    ) -> StableSink<Action> where Action: WorkflowAction, Action.WorkflowType == WorkflowType {
+        let sink = context.makeSink(of: actionType)
+
+        sinks[ObjectIdentifier(actionType)] = sink
+
+        return StableSink(trampoline: self)
```

Comment:
I wonder if we could pass `sink` into `StableSink.init` here and implement `StableSink.send` using that reference, allowing us to delete `sinks`, `bounce`, and `destination`. That would also avoid retaining any `Workflow.Sink` that is no longer referenced by any `StableSink`

Maintaining `sinks` does add some safety (against [this](https://github.com/square/workflow-swift/blob/4399ddc1c588881b2871603e83b6ab5f4303f219/Workflow/Sources/SubtreeManager.swift#L373-L378), I think?) by ensuring we use the latest `Workflow.Sink` that the `RenderContext` has given us for a given `Action` type. But not total safety, because we're not evicting from `sinks` every sink that wasn't remade during the last `render`.

Is the need for that safety greater when the Rendering holds `StableSink`s rather than functions closing over `Workflow.Sink`s?